bgzf.c

```diff
@@ -182,6 +210,7 @@ BGZF *bgzf_open(const char *path, const char *mode)
         hFILE *fpw;
         if ((fpw = hopen(path, mode)) == 0) return 0;
         fp = bgzf_write_init(mode);
+        if (fp == NULL) return NULL;
```

Comment:
errno should be ENOMEM already if malloc failed in bgzf_write_init.  bgzf_write_init can also fail in deflateInit2.  In that case setting errno=ENOMEM might not be correct.  Setting errno=EINVAL for an invalid mode string (in the existing code for bgzf_open) does make sense though.